### DISCUSSION TITLE
Video Tutorial

### DISCUSSION THREAD

**Discussion by SatisfactoryBug** [streamlink/streamlink#5001] [General]
Do anybody have a Video Tutorial to explain how to --twitch-disable-ads  (https://streamlink.github.io/latest/cli.html#cmdoption-twitch-disable-ads)

Or if somebody can make it simple of Explaining ? like 1. Open Streamlink Twitch GUI 2. etc 3. etc  [Sorry for your time]

Sorry for asking as everything is confusing me of:
https://streamlink.github.io/latest/cli/plugins/twitch.html
https://streamlink.github.io/latest/cli/config.html#plugin-specific-configuration-file
https://streamlink.github.io/latest/cli/config.html#syntax

**Comment by phxn** [comment]
You can add the option 'twitch-disable-ads' to your config file, on Windows this is located at C:\Users\<USER>\AppData\Roaming\streamlink\config.

This won't remove all ads though, it will display a Twitch screen when ads play. If you want to skip ads completely you need to subscribe to Twitch Turbo and add your authentication details in your config.

**Reply by bastimeyer** [reply]
> it will display a Twitch screen when ads play

Wrong. `--twitch-disable-ads` filters out ad segments or the "purple screen" from Streamlink's output, aka. everything that's not part of the "actual" stream, so you'll have gaps in the playback while no regular content is available, which is better than having a stream discontinuity in the stream output, as it can break playback.
https://streamlink.github.io/cli.html#cmdoption-twitch-disable-ads

**Reply by phxn** [reply]
Why am I seeing a purple screen when ads take place?

EDIT: It appears I am on 5.1.0 not 5.1.1. Maybe this is why.

EDIT2: Just got it again on 5.1.1... Purple Twitch screen, "Commercial break in progress".

**Reply by bastimeyer** [reply]
I just had a quick look and there appears to be a new change on Twitch's end since today which sometimes causes ad segments to be incorrectly included in the output. I already have an idea though about the fix. I'll open a new issue, so this doesn't have to be discussed here.

This doesn't change the fact though what this parameter does.

**Reply by phxn** [reply]
Sorry I don't understand then. It seems there is a lot of confusion as to what this command actually does.

"--twitch-disable-ads
_Skip embedded advertisement segments at the beginning or during a stream. Will cause these segments to be missing from the output._"

If I don't include this option, I should see ads, correct? If I do include this option I should not see ads, but I will see (as I'm seeing right now) a purple screen with "Commercial break in progress".

Am I wrong in this respect?

So is this pausing method the new workaround in 5.1.1? Because 5.1.0 was still displaying the purple screen for me and recent discussion implies that is to be expected. So if it has changed it must be a 5.1.1 change.

**Reply by bastimeyer** [reply]
> here is a lot of confusion as to what this command actually does.

No, read the changelog and the docs. It is clearly explained in the docs what this parameter does. In hindsight, this should've never been named "disable" but "filter" (back in 2019), but changing the parameter name would be a breaking change, and adding a second one is pretty much unnecessary. 5.1.0 included a fix a couple of weeks ago which fixed one main issue, as you can see in #4942.

If you're seeing the "purple screen" while `--twitch-disable-ads` is set, then this is a bug.

I've just opened #5006 with data that can reproduce the (new) issue.

**Reply by SatisfactoryBug** [reply]
> You can add the option 'twitch-disable-ads' to your config file, on Windows this is located at C:\Users<USER>\AppData\Roaming\streamlink\config.
> 
> This won't remove all ads though, it will display a Twitch screen when ads play. If you want to skip ads completely you need to subscribe to Twitch Turbo and add your authentication details in your config.

Ok I did goto: C:\Users<USER>\AppData\Roaming\streamlink\
With Notepad ++ open config
but just wondering where should i put --twitch-disable-ads as i seeing so many thing in the config, so If ok with you give me a idea where i should put  --twitch-disable-ads

Thank you

**Reply by phxn** [reply]
on a fresh line, but remove the "--" at the beginning.

Here is a simple config for reference:

```
player="C:\Program Files\MPC-HC\mpc-hc64.exe"
ffmpeg-ffmpeg=C:\Program Files\Streamlink\ffmpeg\ffmpeg.exe
player-no-close
default-stream=best
twitch-disable-ads
```

Now when you want to watch a stream you can type "streamlink twitch.tv/stream" and it'll use the settings from your config.
